perfkitbenchmarker/providers/ibmcloud/ibmcloud_virtual_machine.py

```diff
@@ -0,0 +1,668 @@
+    self.instance_start_failed = False
+    assert status == ibm.RUNNING
+    if status != ibm.RUNNING:
+      logging.error('Instance start failed, status: %s', status)
+      self.instance_start_failed = True
+    self.vm_started = not self.instance_start_failed
+    logging.info('Instance %s status %s', self.vmid, status)
+
+  def _StartInstance(self):
+    """Starts a IBM Cloud VM instance."""
+    cmd = ibm.IbmAPICommand(self)
+    cmd.flags['instanceid'] = self.vmid
+    status = cmd.InstanceStart()
+    logging.info('start_instance_poll: last status is %s', status)
+    self.instance_start_failed = False
+    assert status == ibm.RUNNING
+    if status != ibm.RUNNING:
+      logging.error('Instance start failed, status: %s', status)
+      self.instance_start_failed = True
+    self.vm_started = not self.instance_start_failed
+
+  def _CreateFip(self, name: str):
+    """Creates a VNIC in a IBM Cloud VM instance."""
+    cmd = ibm.IbmAPICommand(self)
+    cmd.flags['instanceid'] = self.vmid
+    vnicid = cmd.InstanceGetPrimaryVnic()
+    cmd.flags['name'] = name
+    cmd.flags['target'] = vnicid
+    logging.info('Creating FIP for instanceid: %s', self.vmid)
+    resp = json.loads(cmd.InstanceFipCreate())
+    if resp:
+      logging.info('FIP create resp: %s', resp)
+      assert resp['address'] != None
+      return resp['address'], resp['id']
+    logging.error('FAILED to create FIP for instance %s', self.vmid)
+    return None, None
+
+  def _DeleteFip(self):
+    """Deletes fip in a IBM Cloud VM instance."""
+    logging.info('Deleting FIP, instanceid: %s, fip address: %s, fip id: %s',
+                 self.vmid, self.fip_address, self.fip_id)
+    cmd = ibm.IbmAPICommand(self)
+    cmd.flags['fip_id'] = self.fip_id
+    cmd.InstanceFipDelete()
+
+  def _FindVnicIdByName(self, vnics: [str], name: str):
+    """Finds vnic by name"""
+    for vnic in vnics:
+      if vnic['name'] == name:
+        return vnic['uid']
+    return None
+
+  def _FindVnicIdBySubnet(self, vnics: [str], subnet: str):
+    """Finds vnic by subnet"""
+    for vnic in vnics:
+      if vnic['subnet']['id'] == subnet:
+        return vnic['id']
+    return None
+
+  def _WaitForIPAssignment(self, networkid: str):
+    """Finds the IP address assigned to the vm."""
+    IPv4Address = '0.0.0.0'
+    count = 0
+    while IPv4Address == '0.0.0.0' and \
+    count * FLAGS.ibmcloud_polling_delay < 240:
+      time.sleep(FLAGS.ibmcloud_polling_delay)
+      count += 1
+      cmd = ibm.IbmAPICommand(self)
+      cmd.flags['instanceid'] = self.vmid
+      logging.info('Looking for IP for instance %s, networkid: %s',
+                   self.vmid, networkid)
+
+      resp = cmd.InstanceShow()
+      for network in resp['network_interfaces']:
+        if network['subnet']['id'] == networkid:
+          IPv4Address = network['primary_ipv4_address']
+          break
+      logging.info('Waiting on ip assignment: %s', IPv4Address)
+    assert IPv4Address != '0.0.0.0'
```

Comment:
Recommend raising error instead of assert

Reply:
OK, I will update.